SFML/SFML
# Require all modules are built before building tests

## Description

Currently the tests must recreate a lot of the configuration complexity that the core library has. If you disable a specific module, it is expected that the tests for that module don't get built or run. This approach would be fine if a given module's tests only depended on that one module (in addition to the test framework). This could in fact be true if we were willing to add `operator<<` to each class we test. It would make our test code significantly simpler and avoid the problems this PR seeks to solve. However because we refuse to do that, we're forced to create a 2nd class library only used for testing called `sfml-test-main` which contains definitions for those `operator<<` functions that let us see the contents of a type when a test assertion fails. These operators have the same inter-module dependencies as the core library. If you get the `operator<<` for `sf::Transform` then you also need the `operator<<` for `sf::Vector2<T>` which is used in `sf::Transform`'s interface. That means `sfml-test-main` also needs to recreate the optional module logic such that when you disable the Graphics module for instance, the corresponding Graphics test utilities don't get built.

Maintaining all this optional module build logic in 2-3 places is difficult. Because these code paths get tested so seldom, they're prone to bugs. Our best solution is to add `operator<<` to the core library thus allowing us to totally sidestep this problem. I think many users would appreciate this as well. Our second best solution is to simply require that all modules be built if anyone wants to build the tests. It's reasonable to assume that a potential contributor who wants to run the tests will also be able and willing to enable all modules.  

## Tasks


## How to test this PR?

Disable any module and observe how the tests are not configured, compiled, or ran.
